### PR TITLE
fix(leaderboard modal): language dropdown resets after reopening lb (ykahveci)

### LINKED ISSUE #5417: Selected language in dropdown resets when hiding and reopening the leaderboard
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

ykahveci

### Account config

_No response_

### Current Behavior

Currently (as of c164b842918df43d09bf13cffe7f2e37fe3199d6), when hiding and reopening the leaderboard after viewing a non-English (like in my example German) leaderboard, the leaderboard of the previous language is shown correctly (the title is also correct), however the dropdown resets to English.

This results in an inconsistent UI because the dropdown and page title show a different language. Switching to the English leaderboard is only possible after switching to another leaderboard first.

### Expected Behavior

The dropdown should still have the current language selected after closing and reopening the leaderboard.

Alternatively, the leaderboard could switch back to a default view when opened (such as always displaying the global leaderboard)

### Steps To Reproduce

1. Open the leaderboard
2. Click 'daily'
3. Select a language other than English (in my example German)
4. Press escape to close the leaderboard
5. Open the leaderboard again

The leaderboard now displays the title and content for the German daily leaderboard, but the selected language in the dropdown is English.

### Environment

- OS: Arch Linux (btw)
- Browser: Chromium
- Browser Version: 124.0.6367.201 (Official Build) Arch Linux (64-bit)

### Anything else?

I would like to try to resolve the bug myself.

Looking at the source code, it looks like it could be a regression caused by 97b2766a9e4b4b84e1006572a64ec343b844e9f3 (due to the dropdown element getting completely destroyed when hiding the leaderboard and assuming the default state when recreated).

If I can manage to set up a local development environment, I'll submit a patch.

### PR DESCRIPTION
**Pull request by ykahveci**
### Checks

- [ ] Adding quotes?
  - [ ] Make sure to include translations for the quotes in the description (or another comment) so we can verify their content.
- [ ] Adding a language or a theme?
  - [ ] If is a language, did you edit `_list.json`, `_groups.json` and add `languages.json`?
  - [ ] If is a theme, did you add the theme.css?
    - Also please add a screenshot of the theme, it would be extra awesome if you do so!
- [X] Check if any open issues are related to this PR; if so, be sure to tag them below.
- [X] Make sure the PR title follows the Conventional Commits standard. (https://www.conventionalcommits.org for more info)
- [X] Make sure to include your GitHub username inside parentheses at the end of the PR title

Closes #5417

### CODE REVIEW

File: frontend/src/ts/elements/leaderboards.ts [hunk 1]
```diff
@@ -691,31 +691,37 @@ export function show(): void {
         {
           value: "english",
           text: "english",
-          selected: true,
+          selected: currentLanguage === "english",
```
Comment (fehmer) [inline frontend/src/ts/elements/leaderboards.ts:694]: what about
```data = ["english","spanish", ...]
.map(lang=> {
          value: lang,
          text: lang
          lang === selected: currentLanguage
});
```

Comment (ykahveci) [reply frontend/src/ts/elements/leaderboards.ts:694]: I agree that would be cleaner, especially the more languages might get added.

I'll patch this

Comment (ykahveci) [reply frontend/src/ts/elements/leaderboards.ts]: I refactored the code snippet, do you have any more requests?

**Review by Miodec — APPROVED** [review]
Thank you

**Pull request merged**